data_model/src/trigger/mod.rs

```diff
@@ -165,6 +163,61 @@ impl Identifiable for Trigger<FilterBox> {
 pub struct Id {
     /// Name given to trigger by its creator.
     pub name: Name,
+    /// DomainId of domain of the trigger.
+    pub domain_id: Option<<Domain as Identifiable>::Id>,
+}
+
+impl Id {
+    /// Constructs a new [`Id`].
+    #[inline]
+    pub const fn new(name: Name, domain_id: <Domain as Identifiable>::Id) -> Self {
+        Self {
+            name,
+            domain_id: Some(domain_id),
+        }
+    }
+}
+
+impl core::fmt::Display for Id {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        if let Some(ref domain_id) = self.domain_id {
+            write!(f, "{}${}", self.name, domain_id)
+        } else {
+            write!(f, "{}", self.name)
+        }
+    }
+}
+
+impl FromStr for Id {
+    type Err = ParseError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if s.is_empty() {
+            return Err(ParseError {
+                reason: "Trigger ID cannot be empty",
+            });
+        }
+
+        let vector: Vec<&str> = s.split('$').collect();
+
+        if vector.len() == 1 {
+            return Ok(Self {
+                name: Name::from_str(vector[0])?,
+                domain_id: None,
+            });
+        }
+
+        if vector.len() != 2 {
```

Comment:
IMO we could check `vector.len() > 2` and put this check above creation of `Id`.
This way we would have more clear logic we check errors first and then actually create `Id`.